tests/test_knn_interp.py

```diff
@@ -0,0 +1,231 @@
+
+def test_init():
+    # make sure we can init the interpolator
+    knn = piff.kNNInterp()
+    knn.build(attr_interp, attr_target)
+
+def test_interp():
+    # logger = piff.config.setup_logger(verbose=3, log_file='test_knn_interp.log')
+    logger = None
+    # make sure we can put in the data
+    star_list = generate_data()
+    knn = piff.kNNInterp(n_neighbors=1)
+    knn.build(attr_interp, attr_target)
+    knn.initialize(star_list, logger=logger)
+
+    # make prediction on first 10 items of star_list
+    star_list_predict = star_list[:10]
+    star_list_predicted = knn.interpolateList(star_list_predict, logger=logger)
+    # also on a single star
+    star_predict = star_list_predict[0]
+    star_predicted = knn.interpolate(star_predict)
+
+    # predicted stars should find their exact partner here, so they have the same data
+    np.testing.assert_array_equal(star_predicted.fit.params, star_predict.fit.params)
+    for attr in attr_interp:
+        np.testing.assert_equal(star_predicted.data[attr], star_predict.data[attr])
+
+    # repeat for a star with its starfit removed
+    star_predict = star_list_predict[0]
+    star_predict.fit = None
+    star_predicted = knn.interpolate(star_predict)
+
+    # predicted stars should find their exact partner here, so they have the same data
+    # removed the fit, so don't check that
+    # np.testing.assert_array_equal(star_predicted.fit.params, star_predict.fit.params)
+    for attr in attr_interp:
+        np.testing.assert_equal(star_predicted.data[attr], star_predict.data[attr])
+
+def test_attr_target():
+    # make sure we can do the interpolation only over certain indices in params
+    # make sure we can put in the data
+    star_list = generate_data()
+    attr_target_one = [1]
+    attr_interp_one = ['focal_y']
+    knn = piff.kNNInterp(n_neighbors=1)
+    knn.build(attr_interp_one, attr_target_one)
+    knn.initialize(star_list)
+
+    # predict
+    star_predict = star_list[0]
+    star_predicted = knn.interpolate(star_predict)
+
+    # predicted stars should find their exact partner here, so they have the same data
+    # but here the fit params are not the same!!
+    np.testing.assert_equal(star_predict.fit.params[attr_target_one[0]], star_predicted.fit.params[0])
+    # we should still have the other interp parameter, however, so look at both!
+    for attr in attr_interp:
+        np.testing.assert_equal(star_predicted.data[attr], star_predict.data[attr])
+
+    # repeat for a star with its starfit removed
+    star_predict = star_list[0]
+    star_predict.fit = None
+    star_predicted = knn.interpolate(star_predict)
+
+    # predicted stars should find their exact partner here, so they have the same data
+    # we should still have the other interp parameter, however, so look at both!
+    for attr in attr_interp:
+        np.testing.assert_equal(star_predicted.data[attr], star_predict.data[attr])
+
+def test_yaml():
+    # using piffify executable
+    pass
```

Comment:
Looks like this wasn't written yet.  Should check that things work when reading this in via the config parsing.